lti_consumer/tests/unit/test_api.py

```diff
@@ -175,6 +175,36 @@ def test_retrieve_existing(self):
         self.assertEqual(LtiConfiguration.objects.all().count(), 1)
         self.assertEqual(lti_config_retrieved, lti_config)
 
+    @patch('lti_consumer.api.hasattr')
+    def test_retrieve_ccx_config(self, mock_hasattr):
+        """
+        Check if the API retrieves a model using the master course location of a CCX location.
+        """
+        master_location = 'block-v1:course+test+2020+type@problem+block@test'
+        ccx_location = Mock(return_value='ccx-block-v1:course+test+2020+type@problem+block@test')
+        ccx_location.to_block_locator.return_value = master_location
+        mock_hasattr.return_value = True
+
+        # Create master course location lti_config.
+        lti_config = LtiConfiguration.objects.create(
```

Comment:
What would happen at the edge case where there is no Master course lti_config and when the xblock consumer is used for a CCX the first time? Is this case even possible to exist?. @kuipumu